File: packages/py/tracing/examples/langchain.py
from langchain_openai import ChatOpenAI
from uselemma_tracing import langchain

lemma_callbacks = [langchain(agent_name="support-agent")]


def call_langchain(user_message: str):
    model = ChatOpenAI(model="gpt-4o", callbacks=lemma_callbacks)
    response = model.invoke(user_message)
    return response.content

File: packages/ts/tracing/examples/langchain.ts
import { ChatOpenAI } from "@langchain/openai";
import { HumanMessage } from "@langchain/core/messages";
import { langChain } from "@uselemma/tracing";

const lemmaCallbacks = [langChain({ agentName: "support-agent" })];

export async function callLangChain(userMessage: string) {
  const model = new ChatOpenAI({
    model: "gpt-4o",
    callbacks: lemmaCallbacks,
  });

  const response = await model.invoke([new HumanMessage(userMessage)]);
  return response.content;
}
